src/StreamSelectLoop.php

```diff
@@ -287,8 +287,28 @@ private function streamSelect(array &$read, array &$write, $timeout)
                 }
             }
 
-            // suppress warnings that occur, when stream_select is interrupted by a signal
-            $ret = @\stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
+            /** @var ?callable $previous */
+            $previous = \set_error_handler(function ($errno, $errstr) use (&$previous) {
+                // suppress warnings that occur when `stream_select()` is interrupted by a signal
+                $eintr = \defined('SOCKET_EINTR') ? \SOCKET_EINTR : 4;
+                if ($errno === \E_WARNING && \strpos($errstr, '[' . $eintr .']: ') !== false) {
+                    return;
+                }
+
+                // forward any other error to registered error handler or print warning
+                return ($previous !== null) ? \call_user_func_array($previous, \func_get_args()) : false;
+            });
+
+            try {
+                $ret = \stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
+                \restore_error_handler();
+            } catch (\Throwable $e) { // @codeCoverageIgnoreStart
+                \restore_error_handler();
```

Comment:
Could be moved to 'finally' block?

Reply:
This package still supports [PHP 5.3](https://github.com/reactphp/event-loop/blob/f84ddde32ed1ad22365b8cb864caba60fcb7970c/composer.json#L29) 🙂

Reply:
😣

Reply:
> :persevere:

We know, and it's on the list of things, when it makes sense, to upgrade to a newer minimum PHP version.